conda/libmesh/build.sh

```diff
@@ -75,6 +75,7 @@ fi
 source $SRC_DIR/configure_libmesh.sh
 export INSTALL_BINARY="${SRC_DIR}/build-aux/install-sh -C"
 METHODS="opt oprof devel dbg" configure_libmesh --prefix=${PREFIX}/libmesh \
+                                                --with-future-timpi-dir=${PREFIX}/libmesh \
```

Comment:
Can you not unify this in configure_libmesh.sh?

Reply:
I'll wait for @lindsayad to chime in, since I stole this commit from him, but yeah, this looks suspicious.  Would making this change only in conda/libmesh/build.sh mean that we do the build right in conda but then the first person to `update_and_rebuild_libmesh.sh` manually discovers that MetaPhysicL communication isn't compiling?

Reply:
I also made this change in the update script. The thing is that the install directory is set using different environment variables here and in the update script. I haven’t looked to see whether this build.sh could use LIBMESH_DIR like the update script uses and which our manual users rely on. If so then this could be moved into configure.sh. Anyway it was easiest for me just to follow the prefix configure option

Reply:
The whole point of that configure script is to make sure that we keep consistent configure flags between conda and user builds; I would prefer that we modify it to support this if possible

Reply:
Let's actually just require a `PREFIX` var in `configure_libmesh.sh` and add:
```
--prefix=${PREFIX} --with-future-timpi-dir=${PREFIX/libmesh}
```

to the configure flags with 

```
PREFIX=${PREFIX}/libmesh METHODS="opt oprof devel dbg" configure_libmesh
```

in `conda/libmesh/build.sh`

and

```
SRC_DIR=${SCRIPT_DIR}/../libmesh PREFIX=${LIBMESH_DIR} configure_libmesh ...
```

in `update_and_rebuild_libmesh.sh`?

Reply:
Ah, I see it now.  So this should work as-is.

As far as these two scripts are concerned, it ought to be fine to add `$LIBMESH_DIR` as yet another "scream and die if this env variable isn't defined" entry at the top of configure_libmesh.sh.  I'll do that now and see if anything in CI here screams.  Are there likely to be any users of configure_libmesh.sh *other than* these two scripts, though?

Reply:
No. It's meant to be used internally. I would probably be okay with removing `SRC_DIR` for `LIBMESH_DIR` with a scream-and-die. I have faith in you making a reasonable change that works cleanly for both install methods

Reply:
@loganharbour - Github didn't update fast enough for me, so I missed your last comment here before making + pushing my commit.  Want to check that out and see what you think?